cylc/flow/task_remote_mgr.py

```diff
@@ -522,6 +523,7 @@ def _file_install_callback(self, ctx, platform, install_target):
             RSYNC_LOG.info(
                 'File installation information for '
                 f'{install_target}:\n{ctx.out}')
+            close_log(RSYNC_LOG)
```

Comment:
I think _file_install_callback gets called once per remote-init so if we are remote-init'ing multiple platforms the logs will get closed multiple times?

Sadly I think because of that this is hard to do.

E.G. if we have three platforms the order of events would look something like this:

1. open logs
2. remote-init x
3. remote-init y
4. remote-init z
5. close logs

Where we need to wait for all remote-inits to succeed before closing the log.

Best option I can think of is to close the file install logs on shutdown and reload. On reload we would also need to open the next file-install log. Potential problem with that is if a remote-init is in progress when the reload is started it will try to log to a closed handler which could cause issues.

Reply:
Yes, good spot. I take it we shouldn't have a file installation log per platform?

Reply:
We could stick them into one file or break them apart into separate ones, both would be valid.

If we were to break them apart into separate files then we no longer need to use `logging` (because we can just `with open(<file>)`, there's only one entry to write per file).

Reply:
(one to raise at the team meet tomorrow)

Reply:
(one per install target seems a good idea to me)

Reply:
Added this in https://github.com/cylc/cylc-flow/issues/4802#issuecomment-1091424254